tokens/src/lib.rs

```diff
@@ -416,7 +430,7 @@ impl<T: Trait> MultiCurrency<T::AccountId> for Module<T> {
 	/// or `can_slash` wasn't used appropriately.
 	fn slash(currency_id: Self::CurrencyId, who: &T::AccountId, amount: Self::Balance) -> Self::Balance {
 		if amount.is_zero() {
-			return amount;
+			return Zero::zero();
```

Comment:
Hmm, why creating a new `Zero` value instead of just return `amount` as zero?

Reply:
https://github.com/paritytech/substrate/blob/master/frame/balances/src/lib.rs#L1037

Refer here. You are right, I'll recover it.